lib/rack/multipart/parser.rb

```diff
@@ -378,7 +376,13 @@ def tag_multipart_encoding(filename, content_type, name, body)
               k.strip!
               v.strip!
               v = v[1..-2] if v.start_with?('"') && v.end_with?('"')
-              encoding = Encoding.find v if k == CHARSET
+              if k == "charset"
+                encoding = begin
+                  Encoding.find v
+                rescue ArgumentError
+                  Encoding::BINARY
```

Comment:
The encoding here is submitted by the attacker.  It's questionable to me that we would use an attacker-chosen encoding at all, but if they submit an encoding that doesn't exist, I think it's best to treat it as binary.  If we don't treat it as binary, we need to rescue the ArgumentError and reraise with a more explicit exception class, so that applications using Rack can handle this as a 400-level error instead of a 500-level error.

One thing the code doesn't currently do, but we may want to consider doing, is checking that the encoding of the name/body is valid, and if it isn't valid, forcing a binary encoding.

Reply:
> If we don't treat it as binary, we need to rescue the ArgumentError and reraise with a more explicit exception class, so that applications using Rack can handle this as a 400-level error instead of a 500-level error.

Pragmatically, this seems like the best option: The request is failed because of an invalid encoding (bad request).